File: frontend/src/app/dashboard/model/dto/chart-options/base-chart-options.dto.ts
import { ChartInformationTypeEnum } from '../chart-creation-options.dto';
import { ChartId } from '../../../../shared/model/id-types';

export abstract class BaseChartOptionsDto {
  id: ChartId;
  title: string;
  chartType: ChartInformationTypeEnum;
  selectedTeamIds?: number[];

  public abstract buildChartOptions();
}

export interface ChartTitle {
  text: string;
  align: 'left';
}

File: backend/burning-okr/burning-okr-app/src/main/java/org/burningokr/dto/dashboard/BaseChartOptionsDto.java
package org.burningokr.dto.dashboard;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import lombok.Data;
import org.burningokr.model.dashboard.ChartInformationTypeEnum;

import java.util.Collection;

@JsonTypeInfo( use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "chartType")
@JsonSubTypes({
  @JsonSubTypes.Type(value = PieChartOptionsDto.class, name = "1"),
  @JsonSubTypes.Type(value = LineChartOptionsDto.class, name = "0")
})
@Data
public abstract class BaseChartOptionsDto {
//  Long id;
  String title;
  int chartType;
  Collection<Long> selectedTeamIds;
}
